Source/Core/Core/PowerPC/BreakPoints.cpp

```diff
@@ -33,7 +77,8 @@ bool BreakPoints::IsTempBreakPoint(u32 address) const
 bool BreakPoints::IsBreakPointBreakOnHit(u32 address) const
 {
   return std::any_of(m_breakpoints.begin(), m_breakpoints.end(), [address](const auto& bp) {
-    return bp.address == address && bp.break_on_hit;
+    return bp.address == address && bp.break_on_hit &&
+           (bp.condition.empty() || EvaluateExpression(bp.condition));
```

Comment:
I wonder if UX would be better if `EvaluateExpression` specifically handled the `.empty()` case. That way you cannot forget about it if a different place also wants to consider `bp.condition` the same way.
It would mean that an empty expression is always `true` though.

Thoughts?

Reply:
You could add an EmptyExpressionValue enum class and have EvaluateExpression take one defaulted to true.

Reply:
I'd like to expose expression evaluation directly to the user at some point, so I think I'll split EvaluateExpression in two, and the second function (EvaluateCondition) will handle the empty string case.